chess/position.go

```diff
@@ -0,0 +1,219 @@
+package chess
+
+// A Position describes an arbitrary chess position.
+type Position struct {
```

Comment:
Should `Position` include the half move clock?

Reply:
I don't see any need to at the moment. It shouldn't be too hard to add in the future, so declined for now.